modal/functions.py

```diff
@@ -1229,16 +1276,16 @@ async def _call_generator(self, args, kwargs):
     async def _call_generator_nowait(self, args, kwargs):
         return await _Invocation.create(self.object_id, args, kwargs, self._client)
 
-    @warn_if_generator_is_not_consumed
-    @live_method_gen
-    @synchronizer.no_input_translation
-    async def map(
+    # note that `map()` is not synchronicity-wrapped, since it accepts executable code in the form of
+    # iterators that we don't want to run inside the synchronicity thread. We delegate to `._map()` with
+    # a safer Queue as input
+    def _map_sync(
         self,
-        *input_iterators,  # one input iterator per argument in the mapped-over function/generator
+        *input_iterators: Iterable[Any],  # one input iterator per argument in the mapped-over function/generator
         kwargs={},  # any extra keyword arguments for the function
         order_outputs: bool = True,  # return outputs in order
-        return_exceptions: bool = False,  # propogate exceptions (False) or aggregate them in the results list (True)
-    ) -> AsyncGenerator[Any, None]:
+        return_exceptions: bool = False,  # propagate exceptions (False) or aggregate them in the results list (True)
+    ) -> Generator[Any, None, None]:
         """Parallel map over a set of inputs.
```

Comment:
Is this docstring still going to appear in the reference docs for `Function.map`?